client/src/unifyfs.c

```diff
@@ -2351,6 +2372,12 @@ static int unifyfs_init(void)
         unifyfs_max_long = LONG_MAX;
         unifyfs_min_long = LONG_MIN;
 
+        /* set our current working directory if user gave us one */
+        cfgval = client_cfg.client_cwd;
+        if (cfgval != NULL) {
+            unifyfs_cwd = strdup(cfgval);
+        }
```

Comment:
Question, to see if it's worth considering; do we know what the UnifyFS mount point is at this point? If so, should we check that `client_cwd` matches the mount point here, and if not then at least print something to the error log if not wanting to fail?

Now that I write it, if according to what you put in the docs is true:
> The value specified in ``cwd`` must match the UnifyFS mount point.

can this just be a `yes|no` flag? If yes, then just prepend what they have set as their mount point to the path in your logic in `unifyfs_normalize_path`? Potentially less user error to worry about.

Not sure if either option would cause issues though if we're eventually on planning to support multiple mount points.

Reply:
That's a good idea about checking the mount point.  Yes, we probably can do that.  I'll look at adding that check.

I wanted people to be able to specify a path in case they want to specify something deeper in their directory space, like cwd=/unifyfs/dir1

Reply:
Bonus points:  Adding that check led to discovery of a bug in out intercept path logic.  With our old check, we would also intercept any paths with extra chars, so mount="/unifyfs" would also intercept any path like "/unifyfstrailingchars".

Reply:
> I wanted people to be able to specify a path in case they want to specify something deeper in their directory space, like cwd=/unifyfs/dir1

Ah, yup, that makes sense.

> With our old check, we would also intercept any paths with extra chars...

Wow, hadn't thought of that. Nice catch!